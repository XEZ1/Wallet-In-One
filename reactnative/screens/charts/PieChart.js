
import React from 'react';
import { StyleSheet, Text, ScrollView, Alert, TouchableOpacity, View } from 'react-native';


import { VictoryPie } from "victory-native";

import data from "./chartData.json"
import { useTheme } from 'reactnative/src/theme/ThemeProvider'

import Icon from 'react-native-vector-icons/AntDesign';


export default function PieChartWallet({ navigation }) {

  const {dark, colors, setScheme} = useTheme();

  const handlePressIn = (event, datapoint) => {
    const dataPoint = data[datapoint.index];
    Alert.alert(dataPoint.x);
  };

  let value = 0;
  data.forEach(jsonObj => {
    value += jsonObj.y;
  });

  return (
    <ScrollView
      contentContainerStyle={{
        flexGrow : 1,
        justifyContent: 'center',
        alignItems: 'center',
        paddingBottom: 20,
        backgroundColor: colors.background,
      }}
      style={styles.container}
    >
      <Text style={[styles.title, {color: colors.text}]}>Wallet-In-One</Text>
      <Text style={[styles.amountText, {color: colors.text}]}>Amount: Â£{value}</Text>
      <TouchableOpacity
        onPress={() => navigation.navigate('Bar Chart')}
        testID="Switch Chart"
      >
        <Text style={styles.button}><Icon name='barchart' size={30}/> Switch Chart</Text>
      </TouchableOpacity>
      <VictoryPie
        data={data}
        innerRadius={70}
        padAngle={3}
        labels={() => null}
        events={[{
          target: "data",
          eventHandlers: {
            onPressIn: handlePressIn
          }
        }]}
        colorScale={["red", "blue", "green", "yellow"]}
      />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    fontWeight: '900',
    fontSize: 50,
    alignSelf: 'center',
    paddingVertical: 10,
  },
  amountText:{
    fontWeight: '900',
    fontSize: 25,
    textAlign: 'left',
  },
  button: {
    backgroundColor: 'red',
    color: 'black',
    width: "60%",
    borderRadius: 25,
    textAlign: 'center',
    fontWeight: 'bold',
    padding: "2%",
    fontSize:  30,
    alignSelf: 'center',
}
});

