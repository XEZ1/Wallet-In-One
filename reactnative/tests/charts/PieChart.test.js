import React from 'react';
import PieChart from '../../screens/charts/chartComponents/pieChart';
import renderer from 'react-test-renderer';


var data = {
    "all": [
        {
          "x": "Credit Card",
          "y": 200
        },
        {
          "x": "Cryptocurrency",
          "y": 50
        },
        {
          "x": "Stocks",
          "y": 75
        },
        {
          "x": "Debit Card",
          "y": 20
        }
    ]
  }

var floatData = {
    "all": [
        {
          "x": "Credit Card",
          "y": 200.50
        },
        {
          "x": "Cryptocurrency",
          "y": 50.39
        },
        {
          "x": "Stocks",
          "y": 75.27
        },
        {
          "x": "Debit Card",
          "y": 20.21
        }
    ]
  }

var colours = ["pink", "turquoise", "lime", "#FA991C"]
const handlePressIn = (event, datapoint)=>{console.log(datapoint)};

describe('<PieChart />', () => {
    it('test 4 label pie chart', () => {
        const pieChart = renderer.create(<PieChart colours={colours} data={data} handlePressIn={handlePressIn} labelCount={4} assetSize={17} numSize={27} />);
        expect(pieChart).toMatchSnapshot();         
      });

    it('test 2 label pie chart', () => {
        const pieChart = renderer.create(<PieChart colours={colours} data={data} handlePressIn={handlePressIn} labelCount={2} assetSize={27} numSize={37} />);
        expect(pieChart).toMatchSnapshot();         
      });

    it('test float values pie chart', () => {
        const pieChart = renderer.create(<PieChart colours={colours} data={floatData} handlePressIn={handlePressIn} labelCount={4} assetSize={17} numSize={27} />);
        expect(pieChart).toMatchSnapshot();         
      });

    it('test pie chart no props', () => {
        const pieChart = renderer.create(<PieChart />).toJSON();
        expect(pieChart).toBeNull();
        expect(pieChart).toMatchSnapshot();
      });
  });