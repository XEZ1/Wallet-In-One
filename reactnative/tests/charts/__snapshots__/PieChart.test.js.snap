// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieChart /> test 2 label pie chart 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;

exports[`<PieChart /> test 4 label pie chart 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;

exports[`<PieChart /> test float values pie chart 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;

exports[`<PieChart /> test pie chart no props 1`] = `null`;
